scico/functional/_tvnorm.py

```diff
@@ -1,87 +1,115 @@
 # -*- coding: utf-8 -*-
-# Copyright (C) 2023 by SCICO Developers
+# Copyright (C) 2023-2024 by SCICO Developers
 # All rights reserved. BSD 3-clause License.
 # This file is part of the SCICO package. Details of the copyright and
 # user license can be found in the 'LICENSE' file distributed with the
 # package.
 
-"""Anisotropic total variation norm."""
+"""Total variation norms."""
 
 from typing import Optional, Tuple
 
 from scico import numpy as snp
 from scico.linop import (
     CircularConvolve,
+    Crop,
     FiniteDifference,
     LinearOperator,
+    Pad,
     VerticalStack,
 )
 from scico.numpy import Array
+from scico.typing import DType, Shape
 
 from ._functional import Functional
-from ._norm import L1Norm
+from ._norm import L1Norm, L21Norm
 
 
-class AnisotropicTVNorm(Functional):
-    r"""The anisotropic total variation (TV) norm.
-
-    The anisotropic total variation (TV) norm computed by
-
-    .. code-block:: python
-
-       ATV = scico.functional.AnisotropicTVNorm()
-       x_norm = ATV(x)
+class TVNorm(Functional):
+    r"""Generic total variation (TV) norm.
 
-    is equivalent to
-
-    .. code-block:: python
-
-       C = linop.FiniteDifference(input_shape=x.shape, circular=True)
-       L1 = functional.L1Norm()
-       x_norm = L1(C @ x)
-
-    The scaled proximal operator is computed using an approximation that
-    holds for small scaling parameters :cite:`kamilov-2016-parallel`.
-    This does not imply that it can only be applied to problems requiring
-    a small regularization parameter since most proximal algorithms
-    include an additional algorithm parameter that also plays a role in
-    the parameter of the proximal operator. For example, in :class:`.PGM`
-    and :class:`.AcceleratedPGM`, the scaled proximal operator parameter
-    is the regularization parameter divided by the `L0` algorithm
-    parameter, and for :class:`.ADMM`, the scaled proximal operator
-    parameters are the regularization parameters divided by the entries
-    in the `rho_list` algorithm parameter.
+    Generic total variation (TV) norm with approximation of the scaled
+    proximal operator :cite:`kamilov-2016-parallel`
+    :cite:`kamilov-2016-minimizing`.
     """
```

Comment:
I'm not familiar with these papers. How universally applicable is this approximation? If it always works, fine. If not, is it clear from the docs that if you want TV-regularized reconstruction you should _not_ use `TVNorm`? Is it confusing to say that it is "is equivalent to `linop.FiniteDifference` when they are not actually interchangeable?

Reply:
The norm is exactly equivalent; it's just the prox that's approximated.